zigzag/classes/opt/temporal/loma/multipermute.py

```diff
@@ -59,19 +59,21 @@ def init(multiset):
 
 
 ## Converts our bespoke linked list to a python list.
-def visit(h):
+def visit(h,original_multiset,unordered_loops):
     o = h
     l = []
     while o is not None:
         l.append(o.value)
         o = o.next
-    return l
+    # place unordered loops at the end
+    return [(name,size) for name,size in original_multiset if name in unordered_loops]+l
```

Comment:
Does this not place the "unordered loops" at the start (i.e. innermost loops) of the permutation? As opposed to "at the end" as you state in the PR?